### DISCUSSION TITLE
met error when build shapely/GEOS, unknown file type: /path/to/cached/prefix/lib/libgeos_c.so

### DISCUSSION THREAD

**Discussion by majinge7846** [emscripten-core/emscripten#19693] [Q&A]
I want to build shapely  2.0.1 (depends GEOS) for web pyodide to use.
[shapely](https://github.com/shapely/shapely)

meet error above
INFO: cc build/temp.emscripten_3_1_27_wasm32-cpython-310/src/c_api.o build/temp.emscripten_3_1_27_wasm32-cpython-310/src/coords.o build/temp.emscripten_3_1_27_wasm32-cpython-310/src/geos.o build/temp.emscripten_3_1_27_wasm32-cpython-310/src/lib.o build/temp.emscripten_3_1_27_wasm32-cpython-310/src/pygeom.o build/temp.emscripten_3_1_27_wasm32-cpython-310/src/strtree.o build/temp.emscripten_3_1_27_wasm32-cpython-310/src/ufuncs.o build/temp.emscripten_3_1_27_wasm32-cpython-310/src/vector.o -L/path/to/cached/prefix/lib -lgeos_c -o build/lib.emscripten_3_1_27_wasm32-cpython-310/shapely/lib.cpython-310-wasm32-emscripten.so
wasm-ld: error: unknown file type: /path/to/cached/prefix/lib/libgeos_c.so

WSL version (Windows Subsystem for Linux)： 1.2.5.0
emscripten version 3_1_27
pyodide version 0.2.3

**Comment by sbc100** [comment]
What is the file type of `/path/to/cached/prefix/lib/libgeos_c.so`?   What does `file /path/to/cached/prefix/lib/libgeos_c.so` say?
